easybuild/easyblocks/n/numpy.py

```diff
@@ -244,8 +245,12 @@ def sanity_check_step(self, *args, **kwargs):
         }
         custom_commands = [
             ('python', '-c "import numpy"'),
-            ('python', '-c "import numpy.core._dotblas"'),  # _dotblas is required for decent performance of numpy.dot()
         ]
+        if LooseVersion(self.version) < LooseVersion("1.10"):
+            # _dotblas is required for decent performance of numpy.dot()
+            custom_commands.append (('python', '-c "import numpy.core._dotblas"'))
+        else:
+            custom_commands.append (('python', '-c "from numpy.core.multiarray import dot"'))
```

Comment:
Maybe checking for certain symbols is an option.

See below (this is of course specific to a build on top of MKL).

Unoptimized:

```
$ nm /tmp/vsc40023/numpytest/lib/python2.7/site-packages/numpy/core/multiarray.so | grep dot
00000000001a5ad0 t array_dot
00000000001adc60 t array_vdot
000000000006eab0 t BOOL_dot
000000000006eae0 t BYTE_dot
00000000004e8020 b cached_npy_dot.3634.0.87
00000000004e8060 b cached_npy_dot.3786.0.74
000000000006ea40 t CDOUBLE_dot
00000000001edbc0 t CDOUBLE_vdot
000000000006e9d0 t CFLOAT_dot
00000000001edb50 t CFLOAT_vdot
000000000006f7b0 t CLONGDOUBLE_dot
00000000001edc30 t CLONGDOUBLE_vdot
000000000006f530 t DATETIME_dot
0000000000099870 t dot_alignment_error
000000000006e8e0 t DOUBLE_dot
000000000006e7f0 t FLOAT_dot
000000000006f6f0 t HALF_dot
000000000006eee0 t INT_dot
000000000006f0c0 t LONG_dot
000000000006f440 t LONGDOUBLE_dot
000000000006f280 t LONGLONG_dot
000000000006f810 t OBJECT_dot
00000000001edc90 t OBJECT_vdot
000000000006ece0 t SHORT_dot
000000000006f610 t TIMEDELTA_dot
000000000006ebe0 t UBYTE_dot
000000000006efd0 t UINT_dot
000000000006f1a0 t ULONG_dot
000000000006f360 t ULONGLONG_dot
000000000006ede0 t USHORT_dot
```

Optimized:

```
$ nm $EBROOTNUMPY/lib/python2.7/site-packages/numpy/core/multiarray.so | grep dot
0000000000209040 t array_dot
0000000000211420 t array_vdot
0000000000252b10 t blas_dot
00000000000d2020 t BOOL_dot
00000000000d2050 t BYTE_dot
00000000018e6fe0 b cached_npy_dot.3634.0.87
00000000018e7020 b cached_npy_dot.3788.0.74
0000000000290800 T cblas_cdotc_sub
00000000002907c0 T cblas_cdotu_sub
000000000028fe40 T cblas_ddot
000000000028f820 T cblas_sdot
000000000028f200 T cblas_zdotc_sub
000000000028f1c0 T cblas_zdotu_sub
0000000000292a30 T cdotc
0000000000292a30 T cdotc_
0000000000294170 T cdotu
0000000000294170 T cdotu_
00000000000d1e50 t CDOUBLE_dot
0000000000251550 t CDOUBLE_vdot
00000000000d1c70 t CFLOAT_dot
0000000000251370 t CFLOAT_vdot
00000000000d2d20 t CLONGDOUBLE_dot
0000000000251720 t CLONGDOUBLE_vdot
00000000000d2aa0 t DATETIME_dot
00000000002912f0 T ddot
00000000002912f0 T ddot_
00000000000fcde0 t dot_alignment_error
00000000000d1a90 t DOUBLE_dot
00000000000d18b0 t FLOAT_dot
00000000000d2c60 t HALF_dot
00000000000d2450 t INT_dot
00000000018e8174 b ___kmpv_zeromkl_blas_cdotc_omp_0
00000000018e8160 b ___kmpv_zeromkl_blas_ddot_omp_0
00000000018e814c b ___kmpv_zeromkl_blas_sdot_omp_0
00000000000d2630 t LONG_dot
00000000000d29b0 t LONGDOUBLE_dot
00000000000d27f0 t LONGLONG_dot
0000000000392f00 T mkl_blas_avx2_cdotu
0000000000392800 T mkl_blas_avx2_xcdotc
00000000003944f0 T mkl_blas_avx2_xddot
0000000000395cb0 T mkl_blas_avx2_xsdot
0000000000392200 T mkl_blas_avx2_zdotc
0000000000391c00 T mkl_blas_avx2_zdotu
000000000035e410 T mkl_blas_avx512_cdotu
000000000030a890 T mkl_blas_avx512_mic_cdotu
00000000003091f0 T mkl_blas_avx512_mic_xcdotc
00000000003089d0 T mkl_blas_avx512_mic_xddot
0000000000305db0 T mkl_blas_avx512_mic_xsdot
0000000000304800 T mkl_blas_avx512_mic_zdotc
0000000000303ef0 T mkl_blas_avx512_mic_zdotu
000000000035cff0 T mkl_blas_avx512_xcdotc
000000000035c8a0 T mkl_blas_avx512_xddot
000000000035a0a0 T mkl_blas_avx512_xsdot
0000000000358bc0 T mkl_blas_avx512_zdotc
00000000003584e0 T mkl_blas_avx512_zdotu
0000000000400900 T mkl_blas_avx_cdotu
0000000000400100 T mkl_blas_avx_xcdotc
000000000043b6d0 T mkl_blas_avx_xddot
0000000000401040 T mkl_blas_avx_xsdot
00000000003ff200 T mkl_blas_avx_zdotc
00000000003fe300 T mkl_blas_avx_zdotu
00000000002bed60 T mkl_blas_cdotc
00000000002cce30 T mkl_blas_cdotc_omp
00000000002e3070 T mkl_blas_cdotu
000000000057cf70 T mkl_blas_cnr_def_cdotu
000000000057c000 T mkl_blas_cnr_def_xcdotc
000000000057ac80 T mkl_blas_cnr_def_xddot
000000000057cb90 T mkl_blas_cnr_def_xsdot
000000000057ba10 T mkl_blas_cnr_def_zdotc
000000000057b420 T mkl_blas_cnr_def_zdotu
00000000002bdb40 T mkl_blas_ddot
00000000002c97d0 T mkl_blas_ddot_omp
0000000000517c00 T mkl_blas_def_cdotu
0000000000516d80 T mkl_blas_def_xcdotc
0000000000515f40 T mkl_blas_def_xddot
00000000005178f0 T mkl_blas_def_xsdot
0000000000516a00 T mkl_blas_def_zdotc
0000000000516680 T mkl_blas_def_zdotu
00000000004842a0 T mkl_blas_mc3_cdotu
0000000000483ed0 T mkl_blas_mc3_xcdotc
0000000000484b10 T mkl_blas_mc3_xddot
00000000004850a0 T mkl_blas_mc3_xsdot
0000000000482f60 T mkl_blas_mc3_zdotc
0000000000483370 T mkl_blas_mc3_zdotu
00000000004da320 T mkl_blas_mc_cdotu
0000000000fb1020 T mkl_blas_mc_cdotu_ctn
00000000004da260 T mkl_blas_mc_xcdotc
0000000000faf980 T mkl_blas_mc_xcdotc_ctn
00000000004ca4a0 T mkl_blas_mc_xddot
00000000004cba60 T mkl_blas_mc_xsdot
00000000004da120 T mkl_blas_mc_zdotc
0000000000fa54c0 T mkl_blas_mc_zdotc_ctn
0000000000fb1af0 T mkl_blas_mc_zdotc_htn
00000000004da080 T mkl_blas_mc_zdotu
0000000000fa42d0 T mkl_blas_mc_zdotu_ctn
0000000000fb1f00 T mkl_blas_mc_zdotu_htn
00000000002bcb00 T mkl_blas_sdot
00000000002c5e00 T mkl_blas_sdot_omp
00000000002ea360 T mkl_blas_xcdotc
00000000002e9360 T mkl_blas_xddot
00000000002e75e0 T mkl_blas_xsdot
00000000002e2d70 T mkl_blas_zdotc
00000000002e2a70 T mkl_blas_zdotu
00000000000d2d80 t OBJECT_dot
0000000000251780 t OBJECT_vdot
00000000002932a0 T sdot
00000000002932a0 T sdot_
00000000000d2250 t SHORT_dot
00000000000d2b80 t TIMEDELTA_dot
00000000000d2150 t UBYTE_dot
00000000000d2540 t UINT_dot
00000000000d2710 t ULONG_dot
00000000000d28d0 t ULONGLONG_dot
00000000000d2350 t USHORT_dot
00000000002923e0 T zdotc
00000000002923e0 T zdotc_
00000000002921c0 T zdotu
00000000002921c0 T zdotu_
```